editoast/src/views/path/path_item_cache.rs

```diff
@@ -79,38 +125,42 @@ impl PathItemCache {
             .collect();
 
         Ok(PathItemCache {
-            uic_to_ops,
-            trigram_to_ops,
-            ids_to_ops,
-            existing_track_ids,
+            ops,
+            uic_to_indices,
+            trigram_to_indices,
+            id_to_index,
             track_ids_to_name,
         })
     }
 
     /// Get the operational points associated with an identifier
     pub fn get_from_id(&self, id: &str) -> Option<&OperationalPointModel> {
-        self.ids_to_ops.get(id)
+        self.id_to_index.get(id).map(|&idx| &self.ops[idx])
     }
 
     /// Get the operational points associated with a trigram
-    pub fn get_from_trigram(&self, trigram: &str) -> Option<&Vec<OperationalPointModel>> {
-        self.trigram_to_ops.get(trigram)
+    pub fn get_from_trigram(&self, trigram: &str) -> Option<Vec<&OperationalPointModel>> {
```

Comment:
wouldn't it be easier to just return the iterator and let the caller decide what to do?


```suggestion
    pub fn get_from_trigram(&self, trigram: &str) -> Option<impl Iterator<Item = &OperationalPointModel>> {
```

Reply:
Done in [ae1017364](https://github.com/OpenRailAssociation/osrd/compare/ae1017364c7e150bfd3de861a4be285227d42395..9342d451d5db7a51f31219e7dde37ebcbbfd0639)